app/components/map/index.js

```diff
@@ -135,31 +141,21 @@ class Map extends Component {
 
   onLayout = () => {
     if (!this.state.alertSelected) {
-      if (this.afterRenderTimer) {
-        clearTimeout(this.afterRenderTimer);
+      if (this.areaFeatures && this.areaFeatures.length > 0) {
+        const area = this.getAreaCoordinates(this.areaFeatures[this.state.index]);
+        this.map.fitToCoordinates(area,
+          { edgePadding: { top: 250, right: 250, bottom: 250, left: 250 }, animated: true });
       }
-      this.afterRenderTimer = setTimeout(() => {
-        if (this.areaFeatures && this.areaFeatures.length > 0) {
-          const area = this.getAreaCoordinates(this.areaFeatures[this.state.index]);
-
-          this.map.fitToCoordinates(area,
-            { edgePadding: { top: 250, right: 250, bottom: 250, left: 250 }, animated: true });
-        }
-      }, 300);
     }
   }
 
+  // TODO: check if we really need to map events
   onRegionChangeComplete = (region) => {
     this.updateRegion(region);
   }
 
   onRegionChange = (region) => {
-    if (this.onRegionChangeTimer) {
-      clearTimeout(this.onRegionChangeTimer);
-    }
-    this.onRegionChangeTimer = setTimeout(() => {
-      this.updateRegion(region);
-    }, 100);
+    this.updateRegion(region);
```

Comment:
@j8seangel ok real solution was to remove the event handlers. Throttle has to be called from the render call because if not, on each triggered event it will create a new throttled function.